### PR TITLE
Put under the MINIO_OPERATOR_RUNTIME the Openshit csr-signer addition

### PR DESCRIPTION
**Pull request by pjuarezd**
We will try to import the `csr-singer` secret and load the CA when the `MINIO_OPERATOR_RUNTIME` exists and is set to `Openshift` (case insensitive)

### CODE REVIEW

File: pkg/common/const.go [hunk 1]
```diff
@@ -24,3 +24,19 @@ const (
 	WebhookAPIBucketService = WebhookAPIVersion + "/bucketsrv"
 	WebhookAPIUpdate        = WebhookAPIVersion + "/update"
 )
+
+const (
+	// OperatorRuntimeEnv tells us which runtime we have. (EKS, Rancher, OpenShift, etc...)
+	OperatorRuntimeEnv = "MINIO_OPERATOR_RUNTIME"
+	// OperatorRuntimeK8s is the default runtime when no specific runtime is set
+	OperatorRuntimeK8s Runtime = "k8s"
```
Comment (jiuker) [inline pkg/common/const.go:32]:
```suggestion
	OperatorRuntimeK8s Runtime = "K8S"
```

File: pkg/controller/operator.go [hunk 2]
```diff
@@ -234,5 +238,5 @@ func (c *Controller) createBuckets(ctx context.Context, tenant *miniov2.Tenant,
 
 // getOperatorDeploymentName Internal func returns the Operator deployment name from MINIO_OPERATOR_DEPLOYMENT_NAME ENV variable or the default name
 func getOperatorDeploymentName() string {
-	return env.Get(OperatorDeplymentNameEnv, DefaultDeploymentName)
+	return env.Get(OperatorDeploymentNameEnv, DefaultDeploymentName)
```
Comment (jiuker) [inline pkg/controller/operator.go:241]:
```suggestion
	return strings.ToUpper(env.Get(OperatorDeploymentNameEnv, DefaultDeploymentName))
```

**Review by jiuker — APPROVED** [review]

File: pkg/utils/utils.go [hunk 3]
```diff
@@ -53,3 +57,24 @@ const (
...
 )
+
+// GetOperatorRuntime Retrieves the runtime from env variable
+func GetOperatorRuntime() common.Runtime {
+	envString := os.Getenv(common.OperatorRuntimeEnv)
+	runtimeReturn := common.OperatorRuntimeK8s
+	if envString != "" {
+		envString = strings.TrimSpace(envString)
+		envString = strings.ToUpper(envString)
+		switch envString {
```
Comment (shtripat) [inline pkg/utils/utils.go:68]: May be we can maintain a `map[string]common.Runtime` and return value as
```
if envString != "" {
	if val, ok := m[envString]; ok {
		runtimeReturn = val
	}
}
```
This way we don't need to keep adding cases later.

Comment (pjuarezd) [reply pkg/utils/utils.go]: Sure, is a good Idea, on place already

**Review by shtripat — APPROVED** [review]
lgtm

**Review by jiuker — CHANGES_REQUESTED** [review]

File: pkg/common/const.go [hunk 4]
```diff
@@ -24,3 +24,27 @@ const (
...
+	// OperatorRuntimeOpenshift is the Openshift runtime flag
+	OperatorRuntimeOpenshift Runtime = "OPENSHIFT"
+	// OperatorRuntimeRancher is the Rancher runtime flag
+	OperatorRuntimeRancher Runtime = "RANCHER"
+)
+
+// Runtimes is a map of the supported Kubernetes runtimes
+var Runtimes = map[string]Runtime{
+	"K8S":       OperatorRuntimeK8s,
+	"EKS":       OperatorRuntimeK8s,
```
[the remark this answers is not in the document]
Comment (pjuarezd) [reply pkg/common/const.go]: good catch!

**Review by shtripat — APPROVED** [review]
lgtm

**Pull request merged**